Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -0,0 +1,860 @@
+    {{3415, 0x4502}, "Dark Super Shot Stealth Elf"},
+    {{3416, 0x0000}, "Shark Shooter Terrafin"},
+    {{3417, 0x0000}, "Bone Bash Roller Brawl"},
+    {{3417, 0x4503}, "Legendary Bone Bash Roller Brawl"},
+    {{3420, 0x0000}, "Big Bubble Pop Fizz"},
+    {{3420, 0x450E}, "Birthday Bash Big Bubble Pop Fizz"},
+    {{3421, 0x0000}, "Lava Lance Eruptor"},
+    {{3422, 0x0000}, "Deep Dive Gill Grunt"},
+    {{3423, 0x0000}, "Turbo Charge Donkey Kong (Nintendo Only)"},
+    {{3423, 0x4502}, "Dark Turbo Charge Donkey Kong (Nintendo Only)"},
+    {{3424, 0x0000}, "Hammer Slam Bowser (Nintendo Only)"},
+    {{3424, 0x4502}, "Dark Hammer Slam Bowser (Nintendo Only)"},
+    {{3425, 0x0000}, "Dive-Clops"},
+    {{3425, 0x450E}, "Missile-Tow Dive-Clops"},
+    {{3426, 0x0000}, "Astroblast"},
+    {{3426, 0x4503}, "Legendary Astroblast"},
+    {{3427, 0x0000}, "Nightfall"},
+    {{3428, 0x0000}, "Thrillipede"},
+    {{3428, 0x450D}, "Eggcited Thrillipede"},
+    {{3500, 0x0000}, "Sky Trophy"},
+    {{3501, 0x0000}, "Land Trophy"},
+    {{3502, 0x0000}, "Sea Trophy"},
+    {{3503, 0x0000}, "Kaos Trophy"},
+};
+
+SkylanderPortalWindow::SkylanderPortalWindow(QWidget* parent) : QWidget(parent)
+{
+  setWindowTitle(tr("Skylanders Manager"));
+  setObjectName(QString::fromStdString("skylanders_manager"));
+  setMinimumSize(QSize(700, 200));
+
+  CreateMainWindow();
+
+  connect(&Settings::Instance(), &Settings::EmulationStateChanged, this,
+          &SkylanderPortalWindow::OnEmulationStateChanged);
+
+  installEventFilter(this);
+
+  OnEmulationStateChanged(Core::GetState());
+};
+
+SkylanderPortalWindow::~SkylanderPortalWindow()
+{
+}
+
+void SkylanderPortalWindow::CreateMainWindow()
+{
+  QVBoxLayout* mainLayout = new QVBoxLayout();
+
+  QGroupBox* checkbox_group = new QGroupBox();
+  QHBoxLayout* checkboxLayout = new QHBoxLayout();
+  checkboxLayout->setAlignment(Qt::AlignHCenter);
+  m_checkbox = new QCheckBox(QString::fromStdString("Emulate Skylander Portal"), this);
+  m_checkbox->setChecked(Config::Get(Config::MAIN_EMULATE_SKYLANDER_PORTAL));
+  connect(m_checkbox, &QCheckBox::toggled, [=](bool checked) { EmulatePortal(checked); });
+  checkboxLayout->addWidget(m_checkbox);
+  checkbox_group->setLayout(checkboxLayout);
+  mainLayout->addWidget(checkbox_group);
+
+  auto add_line = [](QVBoxLayout* vbox) {
+    QFrame* line = new QFrame();
+    line->setFrameShape(QFrame::HLine);
+    line->setFrameShadow(QFrame::Sunken);
+    vbox->addWidget(line);
+  };
+
+  m_group_skylanders = new QGroupBox(tr("Active Portal Skylanders:"));
+  QVBoxLayout* vbox_group = new QVBoxLayout();
+  QScrollArea* scroll_area = new QScrollArea();
+
+  for (auto i = 0; i < MAX_SKYLANDERS; i++)
+  {
+    if (i != 0)
+    {
+      add_line(vbox_group);
+    }
+
+    QHBoxLayout* hbox_skylander = new QHBoxLayout();
+    QLabel* label_skyname = new QLabel(QString(tr("Skylander %1")).arg(i + 1));
+    edit_skylanders.at(i) = new QLineEdit();
```

Comment:
```suggestion
    edit_skylanders[i] = new QLineEdit();
```
(we generally don't use `at()`)